File: src/HelloOgre3D/sandbox/state/IdleState.cpp
#include "IdleState.h"
#include "AgentTypeDef.h"
#include "object/AgentObject.h"

IdleState::IdleState(AgentObject* pAgent) : AgentState(pAgent)
{
	m_stateId = "IdleState";
}

IdleState::~IdleState()
{
}

void IdleState::OnEnter()
{
	m_pAgent->RequestState(SSTATE_IDLE_AIM);
}

void IdleState::OnLeave()
{

}

std::string IdleState::OnUpdate(float dt)
{
	if (m_pAgent->IsMoving())
	{
		m_pAgent->SlowMoving(2.0f);
	}

	auto pInput = m_pAgent->GetInput();
	if (pInput->isKeyDown(OIS::KC_2))
	{
		return "toShoot";
	}
	else if (pInput->isKeyDown(OIS::KC_3))
	{
		return "toMove";
	}
	else if (pInput->isKeyDown(OIS::KC_4))
	{
		return "toDeath";
	}

	return "";
}

File: bin/res/scripts/agent/states/IdleState.lua
-- IdleState --

--local m_pAgent = owner;
AgentState_OnEnter = function(agent)
    print("Info==========>>>IdleOnEnter")
    agent:RequestState(SSTATE_IDLE_AIM);
end

AgentState_OnLeave = function(agent)
    print("Info==========>>>IdleOnEnter")
end

AgentState_OnUpdate = function(agent, deltaTime)
    if agent:IsMoving() then
        agent:SlowMoving(2.0)
    end
    
    local pInput = agent:GetInput();
    if IsNumKeyDown(pInput, 2) then
        return "toShoot";
    elseif IsNumKeyDown(pInput, 3) then
        return "toMove";
    elseif IsNumKeyDown(pInput, 4) then
        return "toDeath";
    end

    return "";
end
